pkg/datastore/row.go

```diff
@@ -59,13 +67,70 @@ func (r *operatorSourceRow) GetPackages() []string {
 // represented by operatorSourceRow.
 // The UID of an OperatorSource object is used as the key to uniquely identify
 // an operator source.
-type operatorSourceRowMap map[types.UID]*operatorSourceRow
+type operatorSourceRowMap struct {
+	lock sync.RWMutex
```

Comment:
Should we consider using a thread safe cache like `go-cache` at this point?

Reply:
My use case does not go beyond that of a standard `map`, `go-cache` seems like an overkill.

Reply:
For the record, `go-cache` is pretty light weight. There are other alternatives too. I would rather we not have our own cache implementation, so please add a story to replace this with an off the shelf library.

Reply:
Right now I don't see any use case(s) that goes beyond standard `map`. In the future if there are new use case(s) that warrant a cache library then we will look into it.

Reply:
I tend to agree here. What value at the moment do we get from `go-cache` that we don't get here? I think a simple lock should suffice in this case

Reply:
At a minimum this should be a concurrent data structure. Otherwise we risk introducing code that directly accesses the cache without grabbing the lock. But I won't argue further about this.

Reply:
This PR addresses the issue of concurrency, we are adding locks to make it so.

Reply:
And hence introducing a chance of accessing the datastore without grabbing the lock which won't happen if we use a thread safe cache or inherently concurrent data structure library. Like I said, I won't argue further about this.